electron/launcher/src/components/Header/Header.js

```diff
@@ -1,20 +1,23 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import logo from '../../assets/small_logo.png';
 import { NavLink, Link } from 'react-router-dom';
-import { clientActions } from '_actions';
-import { patcherActions } from '_actions';
+import { patcherActions, clientActions } from '_actions';
 import { connect } from 'react-redux';
 
 const Header = props => {
 
-  const handleButtonClick = () => {
-    // button is diabled while patching
-    if (props.patch.updateProgress.status !== 'downloading') {
-      props.checkForUpdate();
-      if (props.patch.isLatest) {
-        props.startGameClient();
-      }
+  useEffect(() => {
+    if (props.patch.gameLaunchQueued){
+      props.resetGameLaunchQueue();
+      props.startGameClient();
     }
+  }, [props.patch.gameLaunchQueued]);
+
+  const handleButtonClick = () => {
+    props.checkForUpdate({
```

Comment:
do we still disable button during patching?

Reply:
Button remains disabled the entire time we check for update, including the time patch is being downloaded.